### PR TITLE
Categorys

### PR DESCRIPTION
**Pull request by lukas-mchr**
added category to database
added command /createCategory
added command /getCategory
missing: 
  - use of database in commands
  - giving category as argument when doing a withdraw/transaction

### CODE REVIEW

**Review by lukashert — APPROVED** [review]
Sieht gut aus

File: src/main/java/org/bilanzius/Main.java [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
+import org.bilanzius.persistence.sql.SqliteUserDatabaseService;
+import org.bilanzius.utils.HashedPassword;
 import org.bilanzius.utils.Localization;
 
 import java.util.Scanner;
 
 public class Main {
 
-    public static void main(String[] args) {
+    public static void main(String[] args) throws Exception {
```
Comment (niklaskrauth) [inline src/main/java/org/bilanzius/Main.java:14]: Halt stop wat soll das?! Main throws exception? sus

File: src/main/java/org/bilanzius/Main.java [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
+        // Connect to sqllite database
+        var backend = new SqlBackend();
+        backend.connect();
+
+        // Setup
         Localization localization = Localization.getInstance();
```
Comment (niklaskrauth) [inline src/main/java/org/bilanzius/Main.java:20]: Lukas sagt Antipattern

File: src/main/java/org/bilanzius/services/commands/CreateCategoryCommand.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        }
+
+        if (name != null || budget != null) {
+            return createCategory(name, budget);
+        }
+
+        return localization.getMessage("create_category_usage");
+    }
+
+    private Boolean checkIfCategoryExists(String name) {
```
Comment (niklaskrauth) [inline src/main/java/org/bilanzius/services/commands/CreateCategoryCommand.java:48]: kleines b in boolean

File: src/main/java/org/bilanzius/Main.java (hunk 1 above)
Comment (lukas-mchr) [reply src/main/java/org/bilanzius/Main.java:14]: achso, ja war wegen dem datenbankgedöns da brauch er das

File: src/main/java/org/bilanzius/Main.java (hunk 2 above)
Comment (lukas-mchr) [reply src/main/java/org/bilanzius/Main.java:20]: ?

File: src/main/java/org/bilanzius/services/commands/CreateCategoryCommand.java (hunk 3 above)
Comment (lukas-mchr) [reply src/main/java/org/bilanzius/services/commands/CreateCategoryCommand.java:48]: danke chef

**Pull request merged**